woocommerce/woocommerce-ios
# Order Editing: Add billing address handling

Part of https://github.com/woocommerce/woocommerce-ios/issues/4778, https://github.com/woocommerce/woocommerce-ios/issues/4784.
Closes https://github.com/woocommerce/woocommerce-ios/issues/4798.

## Description

Add ability to edit billing address and save it remotely.

### Technical Details

- Added `AddressType` enum as init arg to view model.
- Added `onEditTapped` callback closure to update `BillingInformationViewController` after save action.
- Updated data prefill logic and remote actions to handle shipping/billing based on `AddressType`.

### Note

I've noticed that **removing** (emptying) of phone or email fields ignored on remote action. Likely because of empty string/nil conversion.
Added card [here](https://github.com/woocommerce/woocommerce-ios/projects/39#card-69457552).

## Testing

1. Open existing order.
2. Tap "View Billing Information" in "Customer" section.
4. Tap "edit" icon on Billing Address row.
5. Update any of fields and tap "Done".
6. Wait for remote action to complete and modal to close automatically.
7. Observe correctly updated billing address.

## Screenshots

<img width=350 src="https://user-images.githubusercontent.com/3132438/134680766-6e14f3e9-446f-4f99-b2d6-82ce3dd86502.png">

---
Update release notes:
